### PR TITLE
RFE: Expose exit code from Terminal's shell #72

### PR DESCRIPTION
**Pull request by schnoddelbotz**
As discussed in issue #72 
- Don't panic if ExitCode() is called before RunLocalShell() - return -1 like os.exec would before wait()
- Add a minimal test for the above
- Add windows support
- Update documentation/README with example usage

### CODE REVIEW

File: term.go [hunk 1]
```diff
@@ -216,6 +218,14 @@ func (t *Terminal) Text() string {
 	return t.content.Text()
 }
 
+// ExitCode returns the exit code from the terminal's shell
```
Comment (andydotxyz) [inline term.go:221]: Looking good thanks, but it omits the info about -1 before the terminal command has exited, can you include it in the docs?

**Review by andydotxyz — CHANGES_REQUESTED** [review]
Good implementation thanks. Just a note about the new API return info.

File: term.go (hunk 1 above)
Comment (schnoddelbotz) [reply term.go:221]: done (in README) - hope it's not too verbose...
Thanks!

Comment (andydotxyz) [reply term.go:221]: I was meaning here in the API documentation. Folk basically never use README for API documentation - this section linked is what the IDE will look up during implementation.

Comment (schnoddelbotz) [reply term.go]: makes sense - sorry I got it wrong - fixed

**Review by andydotxyz — APPROVED** [review]
Thanks so much, great change :)

**Pull request merged**